### PR TITLE
Planks create barricades on doors/windows and can be crowbarred

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request

This allows you to build crude barricades on any kind of door (and windows) using planks.  You can also use a crowbar to remove a plank quickly.  Added screentips to all of these as well.

dreamseeker_ZL83dnN8TN

This also adds a hammering_wood.ogg from sound cloud that is public domain to barricade construction or when you build a wooden wall.

## Why It's Good For The Game

Quality of life since there was never any way for people to actually create crude barricades.  Might even be useful to add resistance vs zombies at a later point.

## Changelog

:cl:
add: Add planks used on doors creates crude barricades.  Can also be used on fulltile windows.  Barricades can now be crowbarred to disassemble them quickly.
soundadd: Add hammering_wood.ogg from soundcloud public domain to barricade and wooden wall construction.
/:cl:

### CODE REVIEW

**Comment by at996** [comment]
I like this idea!
perhaps you could expand it to making barricades on windows as well?
a good example of that is the crude barricades on icebox's wooden shack windows.

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -272,6 +276,23 @@
 	if(!user.combat_mode && istype(I, /obj/item/fireaxe))
 		try_to_crowbar(I, user, FALSE)
 		return TRUE
+	if(!user.combat_mode && istype(I, /obj/item/stack/sheet/mineral/wood))
+		var/obj/item/stack/sheet/mineral/wood/plank = I
+
+		if(!plank.get_amount() >= 2)
```
Comment (Mothblocks) [inline code/game/machinery/doors/door.dm:282]:
```suggestion
		if(plank.get_amount() < 2)
```

This code didn't work.

Also, can you make 2 a define?

File: code/game/machinery/doors/door.dm [hunk 2]
```diff
@@ -272,6 +276,23 @@
...
+			balloon_alert(user, "need two [plank] sheets!")
+			return
+
+		balloon_alert(user, "constructing barricade...")
+		playsound(src, 'sound/items/hammering_wood.ogg', 50, TRUE)
+		if(!do_after(user, 5 SECONDS, target = src) || !plank.use(2) || (locate(/obj/structure/barricade/wooden/crude) in loc))
+			return
+
+		balloon_alert(user, "barricade constructed")
+		// mak sure this gets converted to an .ogg (since it's a .wav) and then credit it to soundcloud (public domain)
```
Comment (Mothblocks) [inline code/game/machinery/doors/door.dm:292]: ?

File: code/game/machinery/doors/door.dm [hunk 3]
```diff
@@ -272,6 +276,23 @@
...
 		return TRUE
+	if(!user.combat_mode && istype(I, /obj/item/stack/sheet/mineral/wood))
+		var/obj/item/stack/sheet/mineral/wood/plank = I
+
+		if(!plank.get_amount() >= 2)
+			balloon_alert(user, "need two [plank] sheets!")
+			return
+
+		balloon_alert(user, "constructing barricade...")
+		playsound(src, 'sound/items/hammering_wood.ogg', 50, TRUE)
```
Comment (Mothblocks) [inline code/game/machinery/doors/door.dm:287]:
```suggestion
		playsound(src, 'sound/items/hammering_wood.ogg', 50, vary = TRUE)
```

File: code/game/machinery/doors/door.dm (hunk 2 above)
Comment (timothymtorres) [reply code/game/machinery/doors/door.dm]: It was a reminder to convert the sound from .wav to .ogg

Removed with https://github.com/tgstation/tgstation/pull/69676/commits/fab5f9bf83832c2a9ff88e16ad9fca9c1e27600c

**Comment by timothymtorres** [comment]
> I like this idea! perhaps you could expand it to making barricades on windows as well? 

dreamseeker_NGl3Gsyvl5

Added it to windows.  Retested and good to go.

**Comment by JohnFulpWillard** [comment]
Could this be an element on doors/windows instead?

**Comment by Wallemations** [comment]
image

**Comment by timothymtorres** [comment]
> Could this be an element on doors/windows instead?

Great idea!  Implemented with https://github.com/tgstation/tgstation/pull/69676/commits/3ba311ed1f31a369d25ba6ba8633ba3634140058

**Comment by timothymtorres** [comment]
This is ready.  I wish there was some way for me to specifically make it resistant to zombie attacks but I think it's a lost cause and too much work.

File: code/datums/elements/can_barricade.dm [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+	examine_texts += span_notice("This looks like it can be barricaded with planks of wood.")
+
+/datum/element/can_barricade/proc/on_start_barricade(atom/source, obj/item/stack/sheet/mineral/wood/plank, mob/living/user, params)
+	SIGNAL_HANDLER
+
+	if(user.combat_mode || !istype(plank) || !istype(user))
+		return
+
+	if(plank.get_amount() < PLANK_BARRICADE_AMOUNT)
+		source.balloon_alert(user, "need two [plank] sheets!")
```
Comment (JohnFulpWillard) [inline code/datums/elements/can_barricade.dm:38]:
```suggestion
		source.balloon_alert(user, "need [PLANK_BARRICADE_AMOUNT] [plank] sheets!")
```

File: code/game/machinery/deployable.dm [hunk 5]
```diff
@@ -79,6 +85,15 @@
 				return
 	return ..()
 
+/obj/structure/barricade/wooden/crowbar_act(mob/living/user, obj/item/tool)
+	balloon_alert(user, "deconstructing barricade...")
+	if(!tool.use_tool(src, user, 2 SECONDS, volume=100))
```
Comment (JohnFulpWillard) [inline code/game/machinery/deployable.dm:90]: default volume is 50, is this loudness necessary?

File: code/game/machinery/deployable.dm [hunk 6]
```diff
@@ -63,6 +60,14 @@
 	bar_material = WOOD
 	var/drop_amount = 3
 
+/obj/structure/barricade/wooden/Initialize(mapload)
+	. = ..()
+
+	if(!(flags_1 & NODECONSTRUCT_1))
```
Comment (JohnFulpWillard) [inline code/game/machinery/deployable.dm:66]: Why are we checking for this flag if crowbar_act doesn't check for any of this and works regardless of it?

File: code/game/machinery/deployable.dm (hunk 5 above)
Comment (timothymtorres) [reply code/game/machinery/deployable.dm]: Applied with https://github.com/tgstation/tgstation/pull/69676/commits/5a7118899455875cbe0529e69f9117aa6055e92a

File: code/game/machinery/deployable.dm (hunk 6 above)
Comment (timothymtorres) [reply code/game/machinery/deployable.dm]: Fixed with https://github.com/tgstation/tgstation/pull/69676/commits/2e650bcfcb7f285fe3d9bc4e4019c8d61d7157ba

I saw there was a check for it in:

```
/obj/structure/barricade/deconstruct(disassembled = TRUE)
	if(!(flags_1 & NODECONSTRUCT_1))
		make_debris()
	qdel(src)
```

So I assumed there were some barricades that shouldn't be deconstructable.

**Pull request merged**